### DISCUSSION TITLE
No camera signal on ESP32-CAM-MB

### DISCUSSION THREAD

**Discussion by miko007** [arendst/Tasmota#20176] [Support request]
### PROBLEM DESCRIPTION
I flashed the newest `tasmota-webcam.bin` onto an *ESP32-CAM-MB* module by vendor *AITRIP*, that i bought from amazon. It has an `ESP32-s` chip and an `OV2640` camera module.

I already have plenty of *ESP-32-CAM* modules running tasmota, those all work like a charm. I bought this specific variant, because it comes with a carrier board, providing ftdi and usb power, which seemed to spare me from the hassle of attaching an uart-converter etc.

Tasmota is running, i can connect to the webserver, i can toggle the led an set its brightness etc. but i can not get any camera image whatsoever.

Trying to run `WcInit` by hand yields:
```
00:00:06.464 CMD: wcinit
00:00:06.472 CAM: InitErr 0x103 try 1
00:00:06.600 CAM: InitErr 0xffffffff try 2
00:00:06.728 CAM: InitErr 0xffffffff try 3
00:00:06.728 CAM: InitErr 0xffffffff
00:00:06.858 CAM: InitErr 0xffffffff try 1
00:00:06.986 CAM: InitErr 0xffffffff try 2
00:00:07.114 CAM: InitErr 0xffffffff try 3
00:00:07.114 CAM: InitErr 0xffffffff
00:00:07.116 RSL: RESULT = {"WCInit":"Done"}
````

So i guess, something must be up with the pinout, but i could not find any datasheet for this board and from the looks of it, it seems identical to my other *ESP32-CAM*  boards. The only difference is the carrier board, but the problem persists, even if the carrier board is removed.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): ESP32-CAM-MB
- [X] Tasmota binary firmware version number used: `13.0.2`
  - [X] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: Web Flasher, NodeMCU PyFlasher
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
00:00:04.965 CMD: Backlog Template; Module; GPIO 255
00:00:04.978 RSL: RESULT = {"NAME":"AITHINKER CAM","GPIO":[4992,1,672,1,416,5088,1,1,1,6720,736,704,1,1,5089,5090,0,5091,5184,5152,0,5120,5024,5056,0,0,0,0,4928,576,5094,5095,5092,0,0,5093],"FLAG":0,"BASE":2}
00:00:05.227 RSL: RESULT = {"Module":{"0":"AITHINKER CAM"}}
00:00:05.653 WIF: Connecting to AP1 Propeller Security in mode 11n as tasmota-CDEF3C-3900...
00:00:05.883 RSL: RESULT = {"GPIO0":{"4992":"CAM_XCLK"},"GPIO1":{"0":"None"},"GPIO2":{"672":"SPI MISO1"},"GPIO3":{"0":"None"},"GPIO4":{"416":"PWM1"},"GPIO5":{"5088":"CAM_DATA1"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"6720":"SDCard CS"},"GPIO14":{"736":"SPI CLK1"},"GPIO15":{"704":"SPI MOSI1"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"5089":"CAM_DATA2"},"GPIO19":{"5090":"CAM_DATA3"},"GPIO20":{"0":"None"},"GPIO21":{"5091":"CAM_DATA4"},"GPIO22":{"5184":"CAM_PCLK"},"GPIO23":{"5152":"CAM_HREF"},"GPIO24":{"0":"None"},"GPIO25":{"5120":"CAM_VSYNC"},"GPIO26":{"5024":"CAM_SIOD"},"GPIO27":{"5056":"CAM_SIOC"},"GPIO32":{"4928":"CAM_PWDN"},"GPIO33":{"576":"LedLink_i"},"GPIO34":{"5094":"CAM_DATA7"},"GPIO35":{"5095":"CAM_DATA8"},"GPIO36":{"5092":"CAM_DATA5"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"5093":"CAM_DATA6"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
00:00:14.263 CMD: status 0
00:00:14.266 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"prop-office-cam","FriendlyName":["Kamera Großraumbüro"],"Topic":"tasmota_CDEF3C","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
00:00:14.300 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32-webcam.bin","RestartReason":"Software reset CPU","Uptime":"0T00:00:10","StartupUTC":"","Sleep":50,"CfgHolder":4617,"BootCount":22,"BCResetTime":"2023-12-05T15:39:21","SaveCount":38}}
00:00:14.334 RSL: STATUS2 = {"StatusFWR":{"Version":"13.2.0.3(webcam)","BuildDateTime":"2023-12-05T12:52:06","Core":"2_0_14","SDK":"4.4.6.231122","CpuFrequency":240,"Hardware":"ESP32-D0WDQ6-V3 v3.1","CR":"461/699"}}
00:00:14.347 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Propeller Security",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A00192800000000","00000080","00006000","00004000","00000000"]}}
00:00:14.617 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":1232,"Free":1647,"Heap":143,"StackLowMark":4,"PsrMax":4093,"PsrFree":3337,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":80,"FlashMode":"DIO","Features":["0809","0F0005CE","00040001","00000001","00000000","00000880","80000080","00200000","54000008","00000080","00000000"],"Drivers":"1,2,4,9,10,12,29,38,50,52,62,81","Sensors":"1,127","I2CDriver":""}}
00:00:14.652 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-CDEF3C-3900","IPAddress":"0.0.0.0","Gateway":"10.7.7.1","Subnetmask":"255.255.255.0","DNSServer1":"0.0.0.0","DNSServer2":"0.0.0.0","Mac":"08:F9:E0:CD:EF:3C","IP6Global":"","IP6Local":"","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0}}
00:00:14.676 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_CDEF3C","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
00:00:14.700 RSL: STATUS7 = {"StatusTIM":{"UTC":"1970-01-01T00:00:14","Local":"1970-01-01T00:00:14","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+00:00","Sunrise":"20:13","Sunset":"05:47"}}
00:00:14.725 RSL: STATUS10 = {"StatusSNS":{"Time":"1970-01-01T00:00:14"}}
00:00:14.730 RSL: STATUS11 = {"StatusSTS":{"Time":"1970-01-01T00:00:14","Uptime":"0T00:00:10","UptimeSec":10,"Heap":143,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":27,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":31},"POWER":"OFF","Dimmer":100,"Fade":"OFF","Speed":1,"LedTable":"ON"}}
00:00:14.755 RSL: STATUS12 = {"StatusSTK":{"Exception":28,"Reason":"LoadProhibited","EPC":"400fbeff","EXCVADDR":"000000fc","CallChain":["400fbefc","4011a661","4011a67d","4011b2ba","400d257e","4011a495","40103cfd","401020f1","40121b96","4012cfd6","400df039"]}}


```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
00:00:06.464 CMD: wcinit
00:00:06.472 CAM: InitErr 0x103 try 1
00:00:06.600 CAM: InitErr 0xffffffff try 2
00:00:06.728 CAM: InitErr 0xffffffff try 3
00:00:06.728 CAM: InitErr 0xffffffff
00:00:06.858 CAM: InitErr 0xffffffff try 1
00:00:06.986 CAM: InitErr 0xffffffff try 2
00:00:07.114 CAM: InitErr 0xffffffff try 3
00:00:07.114 CAM: InitErr 0xffffffff
00:00:07.116 RSL: RESULT = {"WCInit":"Done"}


```

### TO REPRODUCE
Install Tasmota onto this specific board an try to access the camera.

### EXPECTED BEHAVIOUR
Camera stream is visible im WebUI and at `http://ip:81/stream`

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Ditch the USB board. It is garbage. Will never work since there is capacitive load on GPIO 0 when left connected. Since this is the clock input pin (20 Mhz!!) that will end in non working Camera.
On Amazon you will find many entrys from buyers which never got it going.
Solution flash and remove the board. Power the cam with a GOOD power source and it will work.
If it still does not work try the Webcam driver variant 2. It tries more often to get crappy hardware going.

**Reply by miko007** [reply]
Well, i understand that. But as i stated above, the removal of the carrier board sadly does not change a thing. The `ESP-32-CAM` module is now running standalone from a 5V 12A power supply, but the mentioned problem persists.

So i would rule the carrier out as the cause of the problem. Also, the thing came pre-flashed with some basic Web UI that showed the camera image and provided some controls *(like brightness etc.)*, that all seemed to work. it is just tasmota not working.

I really have to assume, it has to be something simple, like differences in the pinout or something. As i do not have the source code of that mentioned pre-flashed sketch, i unfortunately can not double check the pins in that, so i am relying on people here that have more insight into the inner workings of tasmota than me.

**Comment by Jason2866** [comment]
No, very simple. Bad hardware. You may have a chance when you swap the camera module against a known good one. If this does not help -> Trash.

**Reply by miko007** [reply]
but how can the hardware be bad, if it works with the pre-flashed sketch? i saw the image of the camera on there, it even was relatively responsive on high resolution...

also, the camera module seems to be confirmed working on the `M5Stack ESP32CAM` board.

**Comment by Jason2866** [comment]
You can try the alternative Tasmota Webcam driver V2 with:
```
[env:tasmota32-webcamv2]
extends                 = env:tasmota32_base
board                   = esp32-fix
board_build.f_cpu       = 240000000L
build_flags             = ${env:tasmota32_base.build_flags}
                          -DCAMERA_MODEL_AI_THINKER
                          -DUSE_WEBCAM
                          -DUSE_WEBCAM_V2
                          -DENABLE_RTSPSERVER
                          -DCODE_IMAGE_STR='"webcam-v2"'
                          -DOTA_URL='""'
```
This driver trys to get bad hardware going with more retrying to start the the cam.

**Reply by miko007** [reply]
sadly, this did not change anything. well, i guess i am throwing those modules to the curb, then. i really hoped, this could be fixed just changing some pin assignments, but by the looks of it, that is not sufficient.

**Comment by Jason2866** [comment]
One thing, in your log there is no valid time seen. Tasmota does not work well without getting time from NTP.

**Comment by Jason2866** [comment]
Here is the official espressif example for the cam. https://github.com/espressif/esp32-camera/tree/master/examples/camera_example
If this fails too, verified bad hardware

**Reply by miko007** [reply]
i now came around to deploy the official example to the board, and it works, no problems. so after all, this has to be an issue with tasmota, doesn't it?

**Reply by Jason2866** [reply]
Maybe yes, maybe no. We can't help since we don't have issues with all cam devices we have.

**Reply by miko007** [reply]
maybe i can send you one of the models to have a look at/play around with it? this seems to be the no. 1 amazon listing for esp32 cam modules, i guess lots of people could have this issue.

i would try to investigate the problem myself, but i would not even know where to start looking. i do not care about the modules, i will just buy other ones for the time being *(as they only cost a few bucks)* but i really want to help making tasmota better for everyone.

**Reply by Jason2866** [reply]
If you send me one, i will have a look. Can't promise anything and not sending devices back.
Ping me at Tasmota Discord under @Jason2866 for details.

**Comment by Jason2866** [comment]
Check this too https://community.home-assistant.io/t/esphome-esp32-wrover-e-camera-failed-solved/496863/4
